src/Surfnet/Stepup/Configuration/Value/AllowedSecondFactorMap.php

```diff
@@ -0,0 +1,70 @@
+<?php
+
+/**
+ * Copyright 2017 SURFnet B.V.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace Surfnet\Stepup\Configuration\Value;
+
+use Surfnet\StepupBundle\Value\SecondFactorType;
+
+final class AllowedSecondFactorMap
```

Comment:
This object was introduced to simplify the logic in the AllowedSecondFactorListService and to make it possible to return an AllowedSecondFactorList per institution. I'm not sure about alternative approaches, but I agree this solution should not be part of the domain.